Source/Evolve/WarpXEvolve.cpp

```diff
@@ -309,6 +317,15 @@ WarpX::Evolve (int numsteps)
             ExecutePythonCallback("afterEsolve");
         }
 
+        if (synchronize_velocity_for_diagnostics &&
+            (multi_diags->DoComputeAndPack(step) || reduced_diags->DoDiags(step) ||
```

Comment:
Please see suggestion above.
Please assign the logic behind `multi_diags->DoComputeAndPack(step) || reduced_diags->DoDiags(step)` to a named boolean as well, because otherwise this `if` gets a bit involved to read.

Reply:
Better?